token/jwt/jwt.go

```diff
@@ -30,17 +30,19 @@ import (
 	"fmt"
 	"strings"
 
+	"context"
+
 	"github.com/dgrijalva/jwt-go"
 	"github.com/ory/fosite"
 	"github.com/pkg/errors"
 )
 
 type JWTStrategy interface {
-	Generate(claims jwt.Claims, header Mapper) (string, string, error)
-	Validate(token string) (string, error)
+	Generate(ctx context.Context, claims jwt.Claims, header Mapper) (string, string, error)
+	Validate(ctx context.Context, token string) (string, error)
+	Hash(ctx context.Context, in []byte) ([]byte, error)
+	Decode(ctx context.Context, token string) (*jwt.Token, error)
 	GetSignature(token string) (string, error)
```

Comment:
Should this take in `context`?

Reply:
Could you elaborate a bit more on the possible signature computation by a strategy? From my understanding the implementation of this method is just parsing a string and should never result in a db lookup or expensive computation.